packages/brick_offline_first/lib/src/offline_queue/request_sqlite_cache_manager.dart

```diff
@@ -74,64 +83,59 @@ class RequestSqliteCacheManager {
-        `$HTTP_JOBS_URL_COLUMN` TEXT,
-        `$HTTP_JOBS_CREATED_AT_COLUMN` INTEGER DEFAULT 0
-      );
-    ''';
-    final db = await getDb();
-    await db.execute(statement);
-
-    final tableInfo = await db.rawQuery('PRAGMA table_info("$HTTP_JOBS_TABLE_NAME");');
-    final createdAtHasBeenMigrated = tableInfo.any((c) => c['name'] == HTTP_JOBS_CREATED_AT_COLUMN);
-    if (!createdAtHasBeenMigrated) {
-      await db.execute(
-          'ALTER TABLE `$HTTP_JOBS_TABLE_NAME` ADD `$HTTP_JOBS_CREATED_AT_COLUMN` INTEGER DEFAULT 0');
-    }
-  }
+  Future<void> migrate();
 
   /// Discover most recent unprocessed job in database convert it back to an HTTP request.
   /// This method also locks the row to make it idempotent to subsequent processing.
-  Future<http.Request?> prepareNextRequestToProcess() async {
+  Future<List<Map<String, dynamic>>> findNextRequestToProcess() async {
     final db = await getDb();
-    final unprocessedRequests = await db.transaction<List<Map<String, dynamic>>>((txn) async {
+    return await db.transaction<List<Map<String, dynamic>>>((txn) async {
       final latestLockedRequests = await _latestRequest(txn, whereLocked: true);
 
       if (latestLockedRequests.isNotEmpty) {
         // ensure that if the request is longer the 2 minutes old it's unlocked automatically
-        final lastUpdated =
-            DateTime.fromMillisecondsSinceEpoch(latestLockedRequests.first[HTTP_JOBS_UPDATED_AT]);
+        final request = latestLockedRequests.first;
+        final lastUpdated = DateTime.fromMillisecondsSinceEpoch(request[updateAtColumn]);
         final twoMinutesAgo = DateTime.now().subtract(const Duration(minutes: 2));
         if (lastUpdated.isBefore(twoMinutesAgo)) {
-          await RequestSqliteCache.unlockRequest(latestLockedRequests.first, txn);
+          await RequestSqliteCache.unlockRequest(
+            data: request,
+            db: txn,
+            primaryKeyColumn: primaryKeyColumn,
+            lockedColumn: lockedColumn,
+            tableName: tableName,
+          );
         }
         if (serialProcessing) return [];
       }
-
       // Find the latest unlocked request
       final unlockedRequests = await _latestRequest(txn, whereLocked: false);
       if (unlockedRequests.isEmpty) return [];
+
       // lock the latest unlocked request
-      await RequestSqliteCache.lockRequest(unlockedRequests.first, txn);
+      await RequestSqliteCache.lockRequest(
+        data: unlockedRequests.first,
+        db: txn,
+        lockedColumn: lockedColumn,
+        primaryKeyColumn: primaryKeyColumn,
+        tableName: tableName,
+      );
 
       // return the next unlocked request (now locked)
       return unlockedRequests;
     });
+  }
 
-    final jobs = unprocessedRequests.map(RequestSqliteCache.sqliteToRequest).cast<http.Request>();
+  // Override to handle conversation from sqlite to request see
+  // example of usage inside request_sqlite_cache
```

Comment:
You want to avoid referencing implementation method documentation unless absolutely necessary. A terse example (`e.g. SqliteProvider`) is acceptable, but largely the abstract is completely independent of the implementation. 
```suggestion
  /// Builds a client-consumable [_RequestMethod] from SQLite row output
```